mobile/test/kotlin/integration/XdsTest.kt

```diff
@@ -0,0 +1,96 @@
+package test.kotlin.integration
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import io.envoyproxy.envoymobile.AndroidEngineBuilder
+import io.envoyproxy.envoymobile.Engine
+import io.envoyproxy.envoymobile.LogLevel
+import io.envoyproxy.envoymobile.XdsBuilder
+import io.envoyproxy.envoymobile.engine.AndroidJniLibrary
+import io.envoyproxy.envoymobile.engine.JniLibrary
+import io.envoyproxy.envoymobile.engine.testing.TestJni
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class XdsTest {
+  private val appContext: Context = ApplicationProvider.getApplicationContext()
+  private lateinit var engine: Engine
+
+  init {
+    AndroidJniLibrary.loadTestLibrary()
+    JniLibrary.load()
+  }
+
+  @Before
+  fun setUp() {
+    TestJni.startHttp2TestServer()
+    TestJni.initXdsTestServer()
+    val latch = CountDownLatch(1)
+    engine = AndroidEngineBuilder(appContext)
+      .addLogLevel(LogLevel.DEBUG)
+      .setOnEngineRunning {
+        latch.countDown()
+      }
+      .setXds(
+        XdsBuilder(
+          TestJni.getXdsTestServerHost(),
+          TestJni.getXdsTestServerPort(),
+        ).addClusterDiscoveryService()
+      )
+      .build()
+    latch.await()
+    TestJni.startXdsTestServer()
```

Comment:
why do we need to wait for the EM engine to finish initialization before starting the xds test server?

Reply:
This is so that we can get the host and port and pass them to the `XdsBuilder`. If we immediately start the xDS server, the xDS server will time out waiting for the connection to be accepted.

Reply:
ah that's right, makes sense